From 5a26731c3c2d2d5a5f208d8ed0c115a8ca623362 Mon Sep 17 00:00:00 2001
From: Junjie Mao <junjie.mao@enight.me>
Date: Tue, 25 Apr 2017 09:06:02 +0800
Subject: [PATCH 09/16] fs: exclude exofs

Signed-off-by: Junjie Mao <junjie.mao@enight.me>
---
 fs/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/Makefile b/fs/Makefile
index 79f522575cba..d851d991430a 100644
--- a/fs/Makefile
+++ b/fs/Makefile
@@ -122,7 +122,7 @@ obj-$(CONFIG_OCFS2_FS)		+= ocfs2/
 obj-$(CONFIG_BTRFS_FS)		+= btrfs/
 obj-$(CONFIG_GFS2_FS)           += gfs2/
 obj-$(CONFIG_F2FS_FS)		+= f2fs/
-obj-y				+= exofs/ # Multiple modules
+#obj-y				+= exofs/ # Multiple modules
 obj-$(CONFIG_CEPH_FS)		+= ceph/
 obj-$(CONFIG_PSTORE)		+= pstore/
 obj-$(CONFIG_EFIVAR_FS)		+= efivarfs/
-- 
1.9.3

